model/riscv_mem.sail

```diff
@@ -228,26 +228,28 @@ function mem_write_ea (addr, width, aq, rl, con) = {
 }
 
 $ifdef RVFI_DII
-val rvfi_write : forall 'n, 0 < 'n <= max_mem_access . (xlenbits, atom('n), bits(8 * 'n), mem_meta) -> unit effect {wreg}
-function rvfi_write (addr, width, value, meta) = {
+val rvfi_write : forall ('a : Type) 'n, 0 < 'n <= max_mem_access . (xlenbits, atom('n), bits(8 * 'n), mem_meta, MemoryOpResult('a)) -> unit effect {wreg}
+function rvfi_write (addr, width, value, meta, result) = {
   rvfi_mem_data->rvfi_mem_addr() = EXTZ(addr);
   rvfi_mem_data_present = true;
-  /* TODO: report tag bit for capability writes and extend mask by one bit. */
-  if width <= 16 then {
+  /* Log only the memory address (without the value) if the write fails. */
+  let write_failed : bool = match result { MemException(_) => true, _ => false };
+  if write_failed then ()
```

Comment:
Writing it like this is a bit odd, probably better to do:
```
match result {
  MemValue(b, _) => if width <= 16 ...
  MemException(_) => ()
};
```
then it matches the read case

(side note: I have no idea what the bool in the MemoryOpResult is for, both the C and OCaml write_ram implementations unconditionally return true... probably we should at least push that down so it's MemoryOpResult(unit) everywhere and only asserted at the boundary, rather than have it bubble all the way up to the instruction semantics and have them assert. Maybe I'll do that...)

Reply:
Much nicer. I've applied that suggestion but I'm not quite sure how to indent the code.